enzy_htp/_interface/propka_interface.py

```diff
@@ -0,0 +1,151 @@
+            temp_pdb_path = tmp_file.name
+            PDBParser.save_structure(temp_pdb_path, stru)
+        
+        try:
+            detailed_results = self.run_propka(temp_pdb_path, work_dir)
+            # Convert detailed results to simple dict for backward compatibility
+            simple_results = {}
+            for res_data in detailed_results:
+                key = (res_data["chain_id"], res_data["res_num"])
+                simple_results[key] = res_data["pKa"]
+            return simple_results
+        finally:
+            if os.path.exists(temp_pdb_path):
+                os.unlink(temp_pdb_path)
+
+    def run_propka(self, pdb_path: str, work_dir: str = None) -> List[Dict[str, Any]]:
+        """
+        Main wrapper for PROPKA that calculates pKa values for residues in a given PDB file.
+
+        Args:
+            pdb_path (str): The path to the PDB file.
+            work_dir (str, optional): The working directory for the calculation. If None, uses system SCRATCH directory.
+
+        Returns:
+            List[Dict[str, Any]]: A list of dictionaries containing detailed pKa information 
+                                 for each ionizable residue.
+        """
+        try:
+            # Set default work_dir to SCRATCH if not provided
+            if work_dir is None:
+                work_dir = eh_config.system.SCRATCH_DIR
+                
+            safe_mkdir(work_dir)
+            original_cwd = os.getcwd()
+            os.chdir(work_dir)
+            
+            # PROPKA calculation
+            options = loadOptions([os.path.abspath(pdb_path)])
+            parameters = read_parameter_file(options.parameters, Parameters())
+            my_molecule = MolecularContainer(parameters, options)
+            my_molecule = read_molecule_file(os.path.abspath(pdb_path), my_molecule)
+            my_molecule.calculate_pka()
+
+            # Extract detailed pKa information
+            conformation = my_molecule.conformations.get('AVR', {})
+            residues_data = []
+            for group in conformation.groups:
+                atom = group.atom
+                row_dict = {}
+                row_dict["res_num"] = atom.res_num
+                row_dict["ins_code"] = atom.icode
+                row_dict["res_name"] = atom.res_name
+                row_dict["chain_id"] = atom.chain_id
+                row_dict["group_label"] = group.label
+                row_dict["group_type"] = getattr(group, "type", None)
+                row_dict["pKa"] = group.pka_value
+                row_dict["model_pKa"] = group.model_pka
+                row_dict["buried"] = group.buried
+                if group.coupled_titrating_group:
+                    row_dict["coupled_group"] = group.coupled_titrating_group.label
+                else:
+                    row_dict["coupled_group"] = None
+                residues_data.append(row_dict)
+
+            return residues_data
+
+        except FileNotFoundError:
```

Comment:
Why is this wrapped in a `try ... except` loop when you could make this check before hand?